Added logging

weateus opened the pull request.

weateus committed:
Added logging

onixred commented on src/main/java/my/pet/ticket/logging/EventType.java:
```diff
@@ -0,0 +1,17 @@
+@Getter
+public enum EventType {
+
+    APP_START("Application started", "Application start error"),
```
можно писать текст на русском. =) (это так для справки)

weateus replied:
Изменил текст сообщений на русский

onixred commented on src/main/java/my/pet/ticket/application/event/ApplicationFailedListener.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void onApplicationEvent(ApplicationFailedEvent event) {
+        Log.ERROR("Handling application start failure event", new EventLog(EventType.APP_INIT));
```
не совсем понятно почему APP_INIT

onixred replied:
он же может завалиться когда сервис уже работает

weateus replied:
Убрал этот тип перечисления

onixred commented on src/main/java/my/pet/ticket/application/event/ApplicationFailedListener.java:
```diff
@@ -0,0 +1,18 @@
+package my.pet.ticket.application.event;
```
по пакетам не факт что это тут

weateus replied:
Вынес в infrastructure.event

onixred commented on src/main/java/my/pet/ticket/logging/DetailLog.java:
```diff
@@ -0,0 +1,24 @@
+
+public class DetailLog implements Detail {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
```
создавать в каждом классе new ObjectMapper не самая эффективная затея. Можно создать класс утилит и в нем статик  ObjectMapper

weateus replied:
Вынес в JsonUtils

onixred commented on src/main/java/my/pet/ticket/logging/EventLog.java:
```diff
@@ -0,0 +1,20 @@
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class EventLog implements Event {
```
нужна ли эта прослойка?   что мешает у EventType добавить интерфейс Event? Создавать каждый раз EventLog не самая лучшая идея.

weateus replied:
Убрал

onixred commented on src/main/java/my/pet/ticket/logging/Log.java:
```diff
@@ -0,0 +1,87 @@
+    private static final String MESSAGE_EVENT_EXCEPTION_DETAILS_TEMPLATE = "Message: '{}'. Event: '{}'. Exception: '{}'. Details: '{}'.";
+
+    public static void DEBUG(String message, Detail... details) {
+        if (LOGGER.isDebugEnabled()) {
```
тут как обычно можно свести к более широкому методу чтобы не делать  в каждом методе реализацию. так как если нужно что-то поправить то  придется исправить в одном месте,  а не в 10 строчках

weateus replied:
Оставил реализацию только в там, где необходимо. В остальных местах добавил вызовы существующих методов

weateus committed:
Review fixes:
- Merged duplicate code in logger
- Removed EventLog
- Created JSON utility class
- Removed ApplicationFailedEvent listener
- Architecture changes
- Changed messages language
